components/telegram_bot_api/sources/channel-updates/channel-updates.mjs

```diff
@@ -40,6 +42,11 @@ export default {
       return;
     }
     const channelPost = body.edited_channel_post ?? body.channel_post;
+
+    if (!channelPost.chat || !channelPost.text) {
+      throw new Error(`Expected body to contain a chat and title, but recieived: ${JSON.stringify(body)}`);
```

Comment:
If `channelPost` is `undefined` when we try to read `channelPost.chat` or `channelPost.text`, I think the same error as before will be thrown:
```
TypeError: Cannot read property 'chat' of undefined
```

We could check for just the `channelPost` instead:
```js
    if (!channelPost) {
      throw new Error(`Expected body to contain a channel_post or edited_channel_post, but recieived: ${JSON.stringify(body)}`);
```

Or, to check that both `channelPost` and `channelPost.chat` are defined:
```js
    if (!channelPost?.chat) {
      throw new Error(`Expected body to contain a chat, but recieived: ${JSON.stringify(body)}`);
```

I don't think we should require that `channelPost.text` is defined because a [`Message`](https://core.telegram.org/bots/api#message) can have an image or audio rather than just text.

Reply:
Love it, much simpler and I didn't know that not about the optional `channelPost.text`.